pontoon/pretranslation/tests/test_tasks.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import absolute_import
+
+from mock import patch
+
+import pytest
+
+from pontoon.base.models import User, Translation
+from pontoon.pretranslation.pretranslate import get_translations
+from pontoon.pretranslation.tasks import pretranslate
+from pontoon.test.factories import (
+    EntityFactory,
+    ResourceFactory,
+    TranslationMemoryFactory,
+    TranslatedResourceFactory,
+    ProjectLocaleFactory,
+)
+
+
+@patch('pontoon.pretranslation.pretranslate.get_google_translate_data')
+@pytest.mark.django_db
+def test_pretranslation_get_translations(gt_mock, locale_b, resource_a, google_translate_locale):
+    entities = [
+        EntityFactory(resource=resource_a, string=x, order=i)
+        for i, x in enumerate(["abaa", "abac", "aaab", "abab"])
+    ]
+
+    entities[1].string_plural = entities[1].string
+    entities[3].string_plural = entities[3].string
+    entities[1].save()
+    entities[3].save()
+
+    google_translate_locale.cldr_plurals = "1, 2"
+    google_translate_locale.save()
+
+    for entity in entities[0:2]:
+        TranslationMemoryFactory.create(
+            entity=entity,
+            source=entity.string,
+            target=entity.string,
+            locale=locale_b,
+        )
+        TranslationMemoryFactory.create(
+            entity=entity,
+            source=entity.string,
+            target=entity.string,
+            locale=google_translate_locale,
+        )
+    # Mock the return value of get_google_translate_data
+    gt_mock.return_value = {
+        'status': True,
+        'translation': entities[2].string,
```

Comment:
Wouldn't it be less confusing if you used `gt_translation` here instead of `entities[2].string`? It took me a while to figure out `response_b` in the last test case.